frontend/src/components/MarkdownText/MarkdownText.tsx

```diff
@@ -23,7 +23,28 @@ export const MarkdownText = ({
   const processedRawString = useMemo(() => {
     // Create new line nodes for every new line in raw string so new lines gets rendered.
     if (multilineBreaks) {
-      return children.replace(/\n/gi, '&nbsp; \n')
+      return (
+        /**
+         * For lines that are contain a token that indents,
+         * we want to remove the whitespace before the newline
+         * so that new lines that come after these tokens
+         * can break out of the indentation group
+         *
+         *  (-|\d+\.|\*): matching character tokens that indents
+         *     -: "-"
+         *     *: "*",
+         *     \d+ : "1.", "2.", etc.
+         *
+         *   \s: whitespace following the token, indentation groups must start with token followed by a whitespace character
+         *
+         *   .*: any character, any number of times, this is the actual text content of the line
+         *
+         *   \n: new line character
+         */
+        children
+          .replace(/\n/gi, '&nbsp; \n')
+          .replace(/(\n(-|\d+\.|\*)\s.*\n)(&nbsp; \n)/gi, '$1 \n')
```

Comment:
Quite a smart approach actually! Wondering why we went with the lookbehind in the first place now. 😄 
nit: the `i` in `/gi` indicates case-insensitive matching, we can actually remove them since newline characters dont have a case.

Reply:
good catch! removed here [bba36a8](https://github.com/opengovsg/FormSG/pull/6941/commits/bba36a895af473036de471b58bfaf9497ff9e1c5)